File: platforms/swift/Sources/ShopifyCheckoutKit/CheckoutLink.swift
import Foundation

/// A link that checkout asked the host app to open.
public struct CheckoutLink: Equatable, Sendable {
    public let url: URL

    public init(url: URL) {
        self.url = url
    }
}

/// The action Checkout Kit should take for a link clicked in checkout.
public enum CheckoutLinkAction: Equatable, Sendable {
    /// Open the link using Checkout Kit's default system behavior.
    case open

    /// Do not open the link.
    case cancel

    /// The app handled the link itself.
    case handled
}

File: platforms/android/lib/src/main/java/com/shopify/checkoutkit/CheckoutLink.kt
package com.shopify.checkoutkit

import android.net.Uri

/** A link that checkout asked the host app to open. */
public class CheckoutLink internal constructor(public val url: Uri)

/** The action Checkout Kit should take for a link clicked in checkout. */
public enum class CheckoutLinkAction {
    /** Open the link using Checkout Kit's default system behavior. */
    Open,

    /** The app handled the link itself. */
    Handled,

    /** Do not open the link. */
    Cancel,
}
